src/components/UIShell/SideNav.js

```diff
@@ -50,12 +50,31 @@ export default class SideNav extends React.Component {
 
   state = {
     isExpanded: true,
+    isFocused: false,
   };
 
   handleExpand = () => {
     this.setState(state => ({ isExpanded: !state.isExpanded }));
   };
 
+  handleFocus = () => {
+    this.setState(state => {
+      if (!state.isFocused) {
```

Comment:
It seems like we’d always want this to be set to true no matter what. Same for handleBlur. Maybe just remove these checks since React won’t rerender unless there’s an update.

Reply:
I thought that setState always triggers an update and causes a rerender, of the virtual DOM if not the actual one. As of [React v16.0 ](https://reactjs.org/blog/2017/09/26/react-v16.0.html#breaking-changes)  returning null will prevent the update.

Reply:
You’re super right, looking good, thanks for make the changes so quickly!